src/lib/repositories_v2/impl/order_status_history_repository_impl.py

```diff
@@ -0,0 +1,112 @@
+        self.session.add(order_status_history)
+        self.session.commit()
+
+    def get_by_id(self, order_status_history_id):
+        return (
+            self.session.query(OrderStatusHistory)
+            .filter(OrderStatusHistory.id == order_status_history_id)
+            .filter(OrderStatusHistory.entity_status == Status.ACTIVE.value)
+            .first()
+        )
+
+    def get_all(self):
+        order_status_histories = self.session.query(OrderStatusHistory).filter(
+            OrderStatusHistory.entity_status == Status.ACTIVE.value
+        )
+        return list(order_status_histories)
+
+    def delete_by_id(self, order_status_history_id, order_status_history):
+        self.session.query(OrderStatusHistory).filter(
+            OrderStatusHistory.id == order_status_history_id
+        ).update(
+            {
+                OrderStatusHistory.entity_status: Status.DELETED.value,
+                OrderStatusHistory.updated_date: datetime.now(),
+                OrderStatusHistory.updated_by: order_status_history.updated_by,
+            }
+        )
+        self.session.commit()
+
+    def update_by_id(self, order_status_history_id, order_status_history):
+        order_status_history_to_be_updated = (
+            self.session.query(OrderStatusHistory)
+            .filter(OrderStatusHistory.id == order_status_history_id)
+            .first()
+        )
+        order_status_history_to_be_updated.order_id = (
+            order_status_history.order_id or order_status_history_to_be_updated.order_id
+        )
+        order_status_history_to_be_updated.from_time = (
+            order_status_history.from_time
+            or order_status_history_to_be_updated.from_time
+        )
+        order_status_history_to_be_updated.to_time = (
+            order_status_history.to_time or order_status_history_to_be_updated.to_time
+        )
+        order_status_history_to_be_updated.from_status = (
+            order_status_history.from_status
+            or order_status_history_to_be_updated.from_status
+        )
+        order_status_history_to_be_updated.to_status = (
+            order_status_history.to_status
+            or order_status_history_to_be_updated.to_status
+        )
+        order_status_history_to_be_updated.updated_date = datetime.now()
+        order_status_history_to_be_updated.updated_by = order_status_history.updated_by
+        self.session.add(order_status_history_to_be_updated)
+        self.session.commit()
+
+    def get_by_order_id(self, order_id):
+        order_status_histories = (
+            self.session.query(OrderStatusHistory)
+            .filter(OrderStatusHistory.entity_status == Status.ACTIVE.value)
+            .filter(OrderStatusHistory.order_id == order_id)
+        )
+        return list(order_status_histories)
+
+    def set_next_status_history_by_order_id(self, order_id, new_status):
+        last_order_status_history_id = self.session.query(
```

Comment:
Could you pick the latest order by date instead of id?

Reply:
done.